### PR TITLE
at least one vowel

### PR DESCRIPTION
**Pull request by kyralee22**
modifies generate letter algorithm to always have one vowel

### CODE REVIEW

**Review by isabellawu90 — DISMISSED** [review]

File: lib/generate_letters.ml [hunk 1]
```diff
@@ -78,10 +78,24 @@ let refill_deck deck =
...
+  | [] -> false
+  | h :: t ->
+      if List.mem h [ 'A'; 'E'; 'I'; 'O'; 'U' ] then true else has_vow t
+
+let insert_vow lst =
+  match lst with
+  | [] -> []
+  | _ :: t -> get_random_letter vowel_amounts :: t
+
+let ensure_vowel lst = if has_vow lst then lst else insert_vow lst
```
Comment (laptou) [inline lib/generate_letters.ml:91]: nit: be consistent w/ naming, `ensure_vowel` is good but then the helper functions should be called `has_vowel` and `insert_vowel`

**Review by laptou — COMMENTED** [review]
mostly lgtm

**Review by laptou — DISMISSED** [review]
lgtm

**Review by owenxr — CHANGES_REQUESTED** [review]
Doesn't ensure vowel on letter-consumption/refill

**Review by owenxr — APPROVED** [review]

File: lib/generate_letters.ml [hunk 2]
```diff
@@ -61,6 +61,19 @@ let realistic_random_letter () =
...
+let rec has_vowel = function
+  | [] -> false
+  | h :: t ->
+      if List.mem h [ 'A'; 'E'; 'I'; 'O'; 'U' ] then true
+      else has_vowel t
+
+let insert_vowel lst =
+  match lst with
+  | [] -> []
+  | _ :: t -> get_random_letter vowel_amounts :: t
```
Comment (laptou) [inline lib/generate_letters.ml:73]: does `insert_vowel` always change the first letter to be a vowel? if so, why does it need a match statement? also, please add a comment explaining how this function works

**Review by laptou — APPROVED** [review]
mostly lgtm

File: lib/generate_letters.ml (hunk 2 above)
Comment (owenxr) [reply lib/generate_letters.ml:73]: > does `insert_vowel` always change the first letter to be a vowel? if so, why does it need a match statement? also, please add a comment explaining how this function works

Looks like it does only change the first letter to be a vowel. I can change it to a map if that's better?

Comment (laptou) [reply lib/generate_letters.ml:73]: it doesn't really matter, i was just wondering why ... you can also leave it as is but please add a comment

Comment (owenxr) [reply lib/generate_letters.ml:77]: Comments added.

Comment (kyralee22) [reply lib/generate_letters.ml:77]: yea it does i didnt think it mattered where the vowel was

**Pull request merged**